src/ayb_db/models.rs

```diff
@@ -2,7 +2,37 @@ use clap::ValueEnum;
 use serde::{Deserialize, Serialize};
 use serde_repr::{Deserialize_repr, Serialize_repr};
 use sqlx::FromRow;
-use std::fmt;
+use std::str::FromStr;
+
+macro_rules! try_from_i16 {
```

Comment:
I love these two macros (though wow macros are hard to read :)). Thank you for introducing them! I left one comment related to pushing the error message and value into the macro implementation below.